crates/paimon/src/io/file_io.rs

```diff
@@ -94,6 +97,13 @@ impl FileIO {
             .into_iter()
             .map(|meta| FileStatus {
                 size: meta.metadata().content_length(),
+                is_dir: meta.metadata().is_dir(),
+                last_modification_time: meta
```

Comment:
When testing, I found that the directory's `last_modification_time` = 0 for a dir. But for the `LocalFileIO` in java it's not. 

![image](https://github.com/user-attachments/assets/d39430ee-c08b-4d39-b398-4822a41d62f9)

Is this a bug for opendal `Fs` service ? CC @Xuanwo

Reply:
I believe it's related to https://github.com/apache/opendal/issues/4615. Would you like to take a look? It might be complex to resolve entirely, but we can start by fixing it for the fs backend first.